router/voteCallbacks.go

```diff
@@ -257,13 +305,136 @@ func (r *Router) getEnvelopeList(request routerRequest) {
 	if request.ListSize == 0 {
 		request.ListSize = MaxListSize
 	}
-	nullifiers := r.vocapp.State.EnvelopeList(request.ProcessID, request.From, request.ListSize, true)
+	envelopes, err := r.Scrutinizer.GetEnvelopes(request.ProcessID)
+	if err != nil {
+		r.sendError(request, fmt.Sprintf("cannot get envelope list: (%s)", err))
+		return
+	}
+	listBytes, err := proto.Marshal(envelopes)
+	if err != nil {
+		r.sendError(request, fmt.Sprintf("cannot marshal vote envelope list: (%s)", err))
+		return
+	}
 	var response types.MetaResponse
-	strnull := []string{}
-	for _, n := range nullifiers {
-		strnull = append(strnull, fmt.Sprintf("%x", n))
+	response.Content = listBytes
+	request.Send(r.buildReply(request, &response))
+}
+
+func (r *Router) getValidatorList(request routerRequest) {
+	var response types.MetaResponse
+	validators, err := r.vocapp.State.Validators(true)
+	if err != nil {
+		r.sendError(request, fmt.Sprintf("cannot get validator list: %v", err))
+		return
 	}
-	response.Nullifiers = &strnull
+	validatorList := new(models.ValidatorList)
+	validatorList.Validators = validators
+	validatorBytes, err := proto.Marshal(validatorList)
+	if err != nil {
+		r.sendError(request, fmt.Sprintf("cannot get validator list: %v", err))
+		return
+	}
+	response.ValidatorList = validatorBytes
+	request.Send(r.buildReply(request, &response))
+}
+
+func (r *Router) getBlock(request routerRequest) {
+	var response types.MetaResponse
+	block := r.Scrutinizer.App.GetBlockByHeight(int64(request.BlockHeight))
+	if block == nil {
+		r.sendError(request, fmt.Sprintf("cannot get block: no block with height %d", request.BlockHeight))
+		return
+	}
+	blockHeader := vochain.MirrorTendermintBlock(block)
+	blockBytes, err := proto.Marshal(blockHeader)
+	if err != nil {
+		r.sendError(request, fmt.Sprintf("cannot get block: %v", err))
+		return
+	}
+	response.Content = blockBytes
```

Comment:
I'm not sure using Protobuf for this case (and also previous) is the best approach. What's the point for not using JSON?

Reply:
Possibly json would make more sense. My reasoning was 
a) use the same encoding for everything
b) smaller payloads
Would be easy for me to refactor as json, though.

Reply:
I'm not sure the cost of encoding/decoding protobuf is woth it the win on size. What do you think @mvdan ?

Reply:
I think we should use JSON, for consistency. This API is exposed by any gateway and could be used either by the Explorer or by a third party application. Having a mixed type API (JSON/Protobuf) is weird.

The work you did with the protobuf types can still be useful in the moment that we decide to migrate the full API from JSON to Protobuf.